dnsmasq.sls

```diff
@@ -24,8 +24,18 @@ dnsmasq:
 #
 # dnsmasq: failed to create listening socket for 172.17.0.1: Cannot assign requested address
 # dnsmasq: unknown interface docker0
-/lib/systemd/system/dnsmasq.service:
-  file.line:
-    - mode: ensure
-    - after: {{ '[Unit]' | regex_escape }}
-    - content: After=docker.service
+{% set dnsmasq_override_file = '/etc/systemd/system/dnsmasq.service.d/override.conf' %}
+dnsmasq_systemd_override:
+  file.managed:
```

Comment:
I'm not entirely sure how systemd overrides work. Will it just add the one setting to the `/lib` file, or will it overwrite the whole thing? I suppose/ hope the former, can you confirm this?

Reply:
The settings are merged. So systemd will first read the file under `/lib` first and then parse the override files and merge the values in. 

> All files with the suffix ".conf" from this directory will be merged in the alphanumeric order and parsed after the main unit file itself has been parsed. This is useful to alter or add configuration settings for a unit, without having to modify unit files.
[Doc](https://www.freedesktop.org/software/systemd/man/systemd.unit.html)

Reply:
Thanks for looking it up! That sounds super nice, then.